base/bucket_n1ql.go

```diff
@@ -63,6 +124,24 @@ func (bucket *CouchbaseBucketGoCB) CreateIndex(indexName string, expression stri
 	return nil
 }
 
+// Waits for index state to be online.  Waits no longer than provided timeout
+func (bucket *CouchbaseBucketGoCB) WaitForIndexOnline(indexName string) error {
+
+	worker := func() (shouldRetry bool, err error, value interface{}) {
+		exists, indexMeta, getMetaErr := bucket.GetIndexMeta(indexName)
+		if exists && indexMeta.State == "online" {
```

Comment:
It would be nice to have a constant defined for `online` and the other possible index meta states.